lib/evaluate/fold.cc

```diff
@@ -156,35 +162,298 @@ ComplexPart FoldOperation(FoldingContext &context, ComplexPart &&complexPart) {
       FoldOperation(context, std::move(complex)), complexPart.part()};
 }
 
-template<typename T>
-Expr<T> FoldOperation(FoldingContext &context, FunctionRef<T> &&funcRef) {
-  ActualArguments args{std::move(funcRef.arguments())};
-  for (std::optional<ActualArgument> &arg : args) {
+namespace intrinsicHelper {
```

Comment:
I don't think that you need a namespace like this in a *.cc file around some types and explicitly `static` functions.

Reply:
yes, I will remove it.